src/formatDate.js

```diff
@@ -8,7 +8,70 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [, , , slider] = fromFormat;
+  const [, , , glue] = toFormat;
+  const [year, month, day] = normalizeDate(
+    date.split(slider),
+    fromFormat,
+    toFormat,
+  );
+
+  const newDate = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i].includes('YY') || toFormat[i].includes('YYYY')) {
+      newDate.push(year);
+      continue;
+    }
+
+    if (toFormat[i].includes('MM')) {
+      newDate.push(month);
+      continue;
+    }
+
+    if (toFormat[i].includes('DD')) {
+      newDate.push(day);
+      continue;
+    }
+  }
+
+  return newDate.join(glue);
 }
 
+function normalizeDate(dateMass, formatMass, toFormat) {
+  const normalDate = [];
+
+  for (let i = 0; i < dateMass.length; i++) {
+    if (formatMass[i].includes('YY') || formatMass[i].includes('YYYY')) {
+      normalDate[0] = dateMass[i];
+      continue;
+    }
+
+    if (formatMass[i].includes('MM')) {
+      normalDate[1] = dateMass[i];
+      continue;
+    }
+
+    if (formatMass[i].includes('DD')) {
+      normalDate[2] = dateMass[i];
+      continue;
+    }
+  }
+
+  if (normalDate[0].length <= 2 && toFormat.includes('YYYY')) {
+    if (+normalDate[0] >= 30) {
+      normalDate[0] = +normalDate[0] + 1900;
+    } else {
+      normalDate[0] = +normalDate[0] + 2000;
+    }
+  } else if (normalDate[0].length > 2 && toFormat.includes('YY')) {
+    if (+normalDate[0] < 2000) {
+      normalDate[0] = +normalDate[0] - 1900;
+    } else {
+      normalDate[0] = +normalDate[0] - 2000;
+    }
+  }
```

Comment:
The logic for converting two-digit years to four-digit years and vice versa may not work correctly for all cases. For example, if the input year is '29', it will be converted to '2029', which might not be the intended behavior. Consider adding more robust logic to handle different century assumptions or provide a way for the user to specify the century.